papaya-core/src/main/java/pl/edu/agh/papaya/repository/SprintRepository.java

```diff
@@ -30,4 +34,11 @@
 
     @Query("select s from Sprint s where s.dateClosed is null or s.dateClosed > ?1")
     List<Sprint> findNotClosed(LocalDateTime localDateTime);
+
+    default Optional<Sprint> findLastInProject(Long projectId) {
+        return OptionalUtil.fromCollection(findInProjectOrderedFromNewest(projectId, PageRequest.of(0, 1)));
```

Comment:
A to jest sprint najbardziej w przyszłości, nie? Późno jest, a myślenie jest trudne, więc właśnie analizowałam kierunki sortowania niepokojąco długo...

Nie wiem jak lepieeej - `findByLatestStartDateInProject`?

Bo "last" też brzmi jak "poprzedni" albo "poprzednio dodany"

Reply:
Btw, tu by robotę zrobiło to takie magiczne `Optional<Sprint> findFirstByProjectIdOrderByDurationPeriodStartDesc(Long projectId);`, ale długa nazwa dosyć wychodzi.